diff --git a/PKGBUILD b/PKGBUILD
new file mode 100644
index 0000000..ab289b6
--- /dev/null
+++ b/PKGBUILD
@@ -0,0 +1,95 @@
+# Maintainer: Sergej Pupykin <pupykin.s+arch@gmail.com>
+# Contributor: Dag Odenhall <dag.odenhall@gmail.com>
+# Contributor: Grigorios Bouzakis <grbzks@gmail.com>
+
+pkgname=slstatus
+pkgver=r540.dd7f189
+pkgrel=1
+pkgdesc='A status monitor for window managers'
+url='http://tools.suckless.org/slstatus'
+arch=('i686' 'x86_64')
+license=('custom:ISC')
+depends=('libx11')
+source=(LICENSE
+        Makefile
+        README
+        arg.h
+        battery.c
+        cpu.c
+        datetime.c
+        disk.c
+        entropy.c
+        hostname.c
+        ip.c
+        kernel_release.c
+        keyboard_indicators.c
+        keymap.c
+        load_avg.c
+        netspeeds.c
+        num_files.c
+        ram.c
+        run_command.c
+        separator.c
+        swap.c
+        temperature.c
+        uptime.c
+        user.c
+        volume.c
+        wifi.c
+        config.def.h
+        config.mk
+        slstatus.1
+        slstatus.c
+        slstatus.h
+        util.c
+        util.h)
+sha256sums=('SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP')
+
+prepare() {
+  cd "$srcdir"
+  cp config.def.h config.h
+}
+
+build() {
+  cd "$srcdir"
+  make X11INC=/usr/include/X11 X11LIB=/usr/lib/X11
+}
+
+package() {
+  cd "$srcdir"
+  make PREFIX=/usr DESTDIR="$pkgdir" install
+  install -Dm644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
+  install -Dm644 README "$pkgdir/usr/share/doc/$pkgname/README"
+}
diff --git a/config.def.h b/config.def.h
index d805331..73e188d 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,7 +1,7 @@
 /* See LICENSE file for copyright and license details. */

 /* interval between updates (in ms) */
-const unsigned int interval = 1000;
+const unsigned int interval = 5000;

 /* text to show if no value can be retrieved */
 static const char unknown_str[] = "n/a";
@@ -14,17 +14,16 @@ static const char unknown_str[] = "n/a";
  *
  * battery_perc        battery percentage              battery name (BAT0)
  *                                                     NULL on OpenBSD/FreeBSD
- * battery_remaining   battery remaining HH:MM         battery name (BAT0)
- *                                                     NULL on OpenBSD/FreeBSD
  * battery_state       battery charging state          battery name (BAT0)
  *                                                     NULL on OpenBSD/FreeBSD
- * cat                 read arbitrary file             path
- * cpu_freq            cpu frequency in MHz            NULL
+ * battery_remaining   battery remaining HH:MM         battery name (BAT0)
+ *                                                     NULL on OpenBSD/FreeBSD
  * cpu_perc            cpu usage in percent            NULL
+ * cpu_freq            cpu frequency in MHz            NULL
  * datetime            date and time                   format string (%F %T)
  * disk_free           free disk space in GB           mountpoint path (/)
  * disk_perc           disk usage in percent           mountpoint path (/)
- * disk_total          total disk space in GB          mountpoint path (/)
+ * disk_total          total disk space in GB          mountpoint path (/")
  * disk_used           used disk space in GB           mountpoint path (/)
  * entropy             available entropy               NULL
  * gid                 GID of current user             NULL
@@ -46,6 +45,7 @@ static const char unknown_str[] = "n/a";
  * ram_total           total memory size in GB         NULL
  * ram_used            used memory in GB               NULL
  * run_command         custom shell command            command (echo foo)
+ * separator           string to echo                  NULL
  * swap_free           free swap in GB                 NULL
  * swap_perc           swap usage in percent           NULL
  * swap_total          total swap size in GB           NULL
@@ -59,11 +59,22 @@ static const char unknown_str[] = "n/a";
  * uptime              system uptime                   NULL
  * username            username of current user        NULL
  * vol_perc            OSS/ALSA volume in percent      mixer file (/dev/mixer)
- *                                                     NULL on OpenBSD/FreeBSD
- * wifi_essid          WiFi ESSID                      interface name (wlan0)
+ *                                                     NULL on OpenBSD
  * wifi_perc           WiFi signal in percent          interface name (wlan0)
+ * wifi_essid          WiFi ESSID                      interface name (wlan0)
  */
 static const struct arg args[] = {
 	/* function format          argument */
-	{ datetime, "%s",           "%F %T" },
+	//{ netspeed_rx, " %s ",    "enp1s0" },
+	//{ netspeed_tx, "%s | ",   "enp1s0" },
+	{ netspeed_rx, " %s ",    "wlp3s0" },
+	{ netspeed_tx, " %s | ",   "wlp3s0" },
+  { load_avg, "%s | ",         NULL },
+  //{ uptime, "Up %s | ",        NULL },
+  { battery_state, "%s",     "BAT0" },
+  { battery_perc, "%s%% ",      "BAT0" },
+  { battery_remaining, "%s | ", "BAT0" },
+	{ run_command, "墳%s | ", "amixer sget Master | tail -1 | awk '{print $4 }' | sed 's@\\(\\[\\|\\]\\)@@g'" },
+	{ datetime, "%s",           "%F %R" },
 };
+
