diff --git a/config.def.h b/config.def.h
index d805331..6bde3e1 100644
--- a/config.def.h
+++ b/config.def.h
@@ -65,5 +65,15 @@ static const char unknown_str[] = "n/a";
  */
 static const struct arg args[] = {
 	/* function format          argument */
-	{ datetime, "%s",           "%F %T" },
+	//{ netspeed_rx, " %s ",    "enp1s0" },
+	//{ netspeed_tx, "%s | ",   "enp1s0" },
+	{ netspeed_rx, " %s ",    "wlp3s0" },
+	{ netspeed_tx, " %s | ",   "wlp3s0" },
+  { load_avg, "%s | ",         NULL },
+  //{ uptime, "Up %s | ",        NULL },
+  { battery_state, "%s",     "BAT0" },
+  { battery_perc, "%s%% ",      "BAT0" },
+  { battery_remaining, "%s | ", "BAT0" },
+	{ run_command, "墳%s | ", "amixer sget Master | tail -1 | awk '{print $4 }' | sed 's@\\(\\[\\|\\]\\)@@g'" },
+	{ datetime, "%s",           "%F %R" },
 };
